packages/jsts/src/rules/S7059/rule.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// https://sonarsource.github.io/rspec/#/rspec/S7059/javascript
+
+import { Rule } from 'eslint';
+import { isRequiredParserServices, generateMeta, isThenable } from '../helpers';
+import * as estree from 'estree';
+import { meta } from './meta';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+const flaggedConstructors = new Set();
+
+export const rule: Rule.RuleModule = {
+  meta: generateMeta(meta as Rule.RuleMetaData, {
+    messages: {
+      noAsyncConstructor: 'Move this asynchronous operation outside of the constructor.',
+    },
+  }),
+  create(context: Rule.RuleContext) {
+    const services = context.sourceCode.parserServices;
+    if (!isRequiredParserServices(services)) {
+      return {};
+    }
+    function isPromiseLike(expr: estree.Expression) {
+      return isRequiredParserServices(services) && isThenable(expr, services);
```

Comment:
why call `isRequiredParserServices(services)` again? you do it already at line 38:

```ts
if (!isRequiredParserServices(services)) {
      return {};
    }
```

```suggestion
      return isThenable(expr, services);
```

Reply:
good catch!